### ISSUE TITLE
azurerm_cdn_frontdoor object id comparisons are case sensitive

### ISSUE THREAD

**Issue by aluchs-hpg** [hashicorp/terraform-provider-azurerm#18615]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.3.1

### AzureRM Provider Version

3.25.0

### Affected Resource(s)/Data Source(s)

azurerm_cdn_frontdoor_rule

### Terraform Configuration Files

```hcl
resource "azurerm_cdn_frontdoor_profile" "profile" {
  name                     = "profile"
  resource_group_name      = "resource_group"
  sku_name                 = "Premium_AzureFrontDoor"
}

resource "azurerm_cdn_frontdoor_rule_set" "ruleSet1" {
  name                     = "ruleSet1"
  cdn_frontdoor_profile_id = azurerm_cdn_frontdoor_profile.profile.id
}

resource "azurerm_cdn_frontdoor_rule" "rule1" {
  name                      = "rule1"
  cdn_frontdoor_rule_set_id = azurerm_cdn_frontdoor_rule_set.ruleSet1.id
  order                     = 0
}
```

### Debug Output/Panic Output

```shell
I would prefer not to since my output contains excessive irrelevant resources and I do not wish to waste the time scrubbing sensitive details. I don't think its necessary, but can provide if absolutely required.
```

### Expected Behaviour

Successive applies following an import of an `azurerm_cdn_frontdoor_rule` resource should not replace the resource on every apply.

### Actual Behaviour

`cdn_frontdoor_rule_set_id` doesn't match because of case sensitivity, resulting in replacing the `azurerm_cdn_frontdoor_rule` resource

```
  # azurerm_cdn_frontdoor_rule.rule1 must be replaced
-/+ resource "azurerm_cdn_frontdoor_rule" "rule1" {
      ~ cdn_frontdoor_rule_set_id   = "/subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourceGroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/ruleSets/ruleSet1" -> "/subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourcegroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/ruleSets/ruleset1" # forces replacement
      ~ cdn_frontdoor_rule_set_name = "rule1" -> (known after apply)
      ~ id                          = "/subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourceGroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/ruleSets/ruleSet1/rules/rule1" -> (known after apply)
        name                        = "rule1"
        # (2 unchanged attributes hidden)

      ~ actions {

          ~ url_redirect_action {
                # (3 unchanged attributes hidden)
            }
        }


        # (2 unchanged blocks hidden)
    }
```

**Please note that this is just one example. I've found this to be true amongst many of the new Front Door Standard/Premium resources including:**
* `azurerm_cdn_frontdoor_endpoint` (on property `cdn_frontdoor_profile_id`)
* `azurerm_cdn_frontdoor_origin_group` (on property `cdn_frontdoor_profile_id`)
* `azurerm_cdn_frontdoor_origin` (on property `cdn_frontdoor_origin_group_id`)
* `azurerm_cdn_frontdoor_route` (on properties `cdn_frontdoor_endpoint_id`, `cdn_frontdoor_origin_group_id`, `cdn_frontdoor_origin_ids`)
* `azurerm_cdn_frontdoor_custom_domain` (on property `cdn_frontdoor_profile_id`)
* Maybe others, I haven't done comprehensive testing, but these for sure.

The only reason I'm using rule as an example is because it persists even across apply and replacement, but none of them should be case sensitive after importing with different than expected case, especially when object IDs are copied from the Azure Portal.

### Steps to Reproduce

1. `terraform import azurerm_cdn_frontdoor_rule_set.ruleSet1 /subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourceGroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/ruleSets/ruleSet1`
2. `terraform import azurerm_cdn_frontdoor_rule.rule1 /subscriptions/01234567-89ab-cdef-0123-456789abcdef/resourceGroups/resourceGroup1/providers/Microsoft.Cdn/profiles/profile1/ruleSets/ruleSet1/rules/rule1`
3. `terraform apply` (get above actual behavior)
4. `terraform apply` (get above actual behavior again, even after applying and replacing the rule)

### Important Factoids

None

### References

None

**Comment by aluchs-hpg** [comment]
Tagging @WodansSon for visibility since they are actively working on this functionality. By the way, thank you so much for all the hard work on the resource types so far, it is really really appreciated.

**Comment by WodansSon** [comment]
@aluchs-hpg, thank you for opening this issue. This issue has a lot of history as there is a long standing bug in Portal where it lower cases the resource ID and has caused many issues in the past with Terraform. That said, I have done what I can in the linked [PR](https://github.com/hashicorp/terraform-provider-azurerm/pull/18600) to mitigate that issue as much as I possibly can within the provider. You need to keep in mind that Terraform is case sensitive by `default` and some of these issues cannot be resolved in the provider itself, but would require a change to the Terraform core runtime. 🚀

**Comment by aluchs-hpg** [comment]
I really appreciate the context, I've used terraform for years, but only recently with Azure, so I'm late to the party on this issue and it just surprised me. Thanks for doing what you can and have - seriously, amazing work. I look forward to trying out the changes when they are released.

**Comment by aluchs-hpg** [comment]
Understood. It makes sense that this is out of scope if it is not an issue limited to these resources. I've been able to mostly workaround by deleting resources and recreating with terraform rather than importing. My biggest concern was having to do this with custom domains because we don't mange our DNS in Azure so it would have been a request to another team, but so far the domain resources have been solved by deleting from state and re-importing with the casing that terraform expects.

I apologize you spent time on it only to throw it away. Thanks again for all the hard work. Looking forward to the multiple routes per domain fix.

If you believe there's nothing remaining to work on this issue, we can close it.
